frontend/public/components/secrets/create-secret.tsx

```diff
@@ -293,24 +248,68 @@ class SSHAuthSubform extends React.Component<SSHAuthSubformProps, SSHAuthSubform
     }, () => this.props.onChange(this.state));
   }
   render() {
-    return <div className="form-group">
-      <label className="control-label" htmlFor="ssh-privatekey">SSH Private Key</label>
-      <div>
-        <FileInput onChange={this.onFileChange} />
-        <p className="help-block">Upload your private SSH key file.</p>
-        <textarea className="form-control co-create-secret-form__textarea"
-          id="ssh-privatekey"
-          name="privateKey"
-          onChange={this.changeData}
-          value={this.state['ssh-privatekey']}
-          aria-describedby="ssh-privatekey-help"
-          required />
-        <p className="help-block" id="ssh-privatekey-help">Private SSH key file for Git authentication.</p>
-      </div>
-    </div>;
+    return <DropableFileInput
+      onChange={this.onFileChange}
+      inputFileData={this.state['ssh-privatekey']}
+      id="ssh-privatekey"
+      label="SSH Private Key"
+      inputFieldHelpText="Drag and drop your private SSH key here or browse to upload it."
+      textareaFieldHelpText="Private SSH key file for Git authentication." />;
+  }
+}
+
+class SourceSecretForm extends React.Component<SourceSecretFormProps, SourceSecretFormState> {
+  constructor (props) {
+    super(props);
+    this.state = {
+      type: this.props.secretType,
+      stringData: this.props.stringData || {},
+    };
+    this.changeAuthenticationType = this.changeAuthenticationType.bind(this);
+    this.onDataChanged = this.onDataChanged.bind(this);
+  }
+  changeAuthenticationType(event) {
+    this.setState({
+      type: event.target.value
+    }, () => this.props.onChange(this.state));
+  }
+  onDataChanged (secretsData) {
+    this.setState({
+      stringData: {...secretsData},
+    }, () => this.props.onChange(this.state));
+  }
+  render () {
+    return <React.Fragment>
+      {this.props.isCreate
+        ? <div className="form-group">
+          <label className="control-label" htmlFor="secret-type" >Authentication Type</label>
+          <div>
+            <select onChange={this.changeAuthenticationType} value={this.state.type} className="form-control" id="secret-type">
+              <option value={SecretType.basicAuth}>Basic Authentication</option>
+              <option value={SecretType.sshAuth}>SSH Key</option>
+            </select>
+          </div>
+        </div>
+        : null
+      }
+      { this.state.type === SecretType.basicAuth
+        ? <BasicAuthSubform onChange={this.onDataChanged} stringData={this.state.stringData} />
+        : <SSHAuthSubform onChange={this.onDataChanged} stringData={this.state.stringData} />
+      }
+    </React.Fragment>;
   }
 }
 
+const secretFormFactory = secretType => {
+  let secretForm = null;
+  if (secretType === SecretTypeAbstraction.webhook){
+    secretForm = withSecretForm(WebHookSecretForm);
+  } else {
+    secretForm = withSecretForm(SourceSecretForm);
+  }
+  return withDragDropContext(secretForm);
```

Comment:
@spadgett wrapping the returned component from the `withSecretForm` HOC into the `withDragDropContext` HOC... not sure we want to load the HTML5Context for all the secret forms though. Or lets load them only those that will actually used DnD ?